src/main/java/org/opensearch/knn/index/fetch/KNNFetchSubPhase.java

```diff
@@ -0,0 +1,226 @@
+                    ContextIndexSearcher searcher = fetchContext.searcher();
+                    final Weight childWeight = searcher.createWeight(searcher.rewrite(childFilter), ScoreMode.COMPLETE_NO_SCORES, 1f);
+
+                    LeafReaderContext subReaderContext = hitContext.readerContext();
+                    Scorer childScorer = childWeight.scorer(subReaderContext);
+                    DocIdSetIterator childIter = childScorer.iterator();
+                    BitSet parentBits = fetchContext.getQueryShardContext().bitsetFilter(parentFilter).getBitSet(subReaderContext);
+
+                    int currentParent = hit.docId() - subReaderContext.docBase;
+                    int previousParent = parentBits.prevSetBit(currentParent - 1);
+                    int childDocId = childIter.advance(previousParent + 1);
+                    SourceLookup nestedVecSourceLookup = new SourceLookup();
+
+                    // when nested field only have vector field and exclude source, list is empty
+                    boolean isEmpty = nestedDocList.isEmpty();
+
+                    for (int offset = 0; childDocId < currentParent && childDocId != DocIdSetIterator.NO_MORE_DOCS; childDocId = childIter
+                        .nextDoc(), offset++) {
+                        nestedVecSourceLookup.setSegmentAndDocument(subReaderContext, childDocId);
+                        List<Object> nestedVecDocValuesSource = f.fetcher.fetchValues(nestedVecSourceLookup);
+                        if (nestedVecDocValuesSource == null || nestedVecDocValuesSource.isEmpty()) {
+                            continue;
+                        }
+                        if (isEmpty) {
+                            nestedDocList.add(new HashMap<String, Object>());
+                        }
+                        if (offset < nestedDocList.size()) {
+                            Object o2 = nestedDocList.get(offset);
+                            if (o2 instanceof Map) {
+                                Map<String, Object> o2map = (Map<String, Object>) o2;
+                                String suffix = f.field.substring(path.length() + 1);
+                                o2map.put(suffix, nestedVecDocValuesSource.get(0));
+                            }
+                        } else {
+                            /**
+                             * TODO nested field partial doc only have vector and source exclude
+                             * this source map nestedDocList would out-of-order, can not fill the vector into right offset
+                             * "nested_field" : [
+                             *    {"nested_vector": [2.6, 2.6]},
+                             *    {"nested_numeric": 2, "nested_vector": [3.1, 2.3]}
+                             *  ]
+                             */
+                            throw new UnsupportedOperationException("Nested Field should not be empty");
```

Comment:
can we add more verbosity to the exception message here.

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1571/commits/56a4e4061c3898850c9f56584bdcf535d5f57978